grayskull/strategy/py_build.py

```diff
@@ -0,0 +1,158 @@
+from conda.models.match_spec import MatchSpec
+from packaging.requirements import Requirement
+from souschef.recipe import Recipe
+
+from grayskull.config import Configuration
+from grayskull.strategy.abstract_strategy import AbstractStrategy
+from grayskull.strategy.pypi import compose_test_section
+
+log = logging.getLogger(__name__)
+
+
+class PyBuild(AbstractStrategy):
+    @staticmethod
+    def fetch_data(recipe: Recipe, config: Configuration, sections=None):
+        project = build.ProjectBuilder(config.name)
+
+        recipe["source"]["path"] = "../"
+        # XXX relative to output. "git_url: ../" is also a good choice.
+
+        with tempfile.TemporaryDirectory(prefix="grayskull") as output:
+            build_system_requires = project.build_system_requires
+            requires_for_build = project.get_requires_for_build("wheel")
+            # If those are already installed, we can get the extras requirements
+            # without invoking pip e.g. setuptools_scm[toml]
+            print("Requires for build:", build_system_requires, requires_for_build)
+
+            # Example of finding extra dependencies for a distribution 'scm' (is
+            # there a dict API?) Subtract "has non-extra marker" dependencies from
+            # this set.
+            #
+            # Easier API is deprecated pkg_resources.Distribution().requires(extras=())
+            #
+            # for e in scm.metadata.get_all('provides-extra'):
+            #   print (e, [x for x in r if x.marker and x.marker.evaluate({'extra':e})])
+            #
+            #     docs [<Requirement('entangled-cli[rich]; extra == "docs"')>, <Requirement('mkdocs; extra == "docs"')>, <Requirement('mkdocs-entangled-plugin; extra == "docs"')>, <Requirement('mkdocs-material; extra == "docs"')>, <Requirement('mkdocstrings[python]; extra == "docs"')>, <Requirement('pygments; extra == "docs"')>]
+            #     rich [<Requirement('rich; extra == "rich"')>]
+            #     test [<Requirement('build; extra == "test"')>, <Requirement('pytest; extra == "test"')>, <Requirement('rich; extra == "test"')>, <Requirement('wheel; extra == "test"')>]
+            #     toml []
+
+            # build the project's metadata "dist-info" directory
+            metadata_path = Path(project.metadata_path(output_directory=output))
+
+            distribution = PathDistribution(metadata_path)
+
+            # real distribution name not pathname
+            config.name = distribution.name  # see also _normalized_name
+            config.version = distribution.version
+
+            # grayskull thought the name was the path. correct that.
+            if recipe[0] == '#% set name = "." %}':  # XXX fragile
+                # recipe[0] = x does not work
+                recipe._yaml._yaml_get_pre_comment()[0].value = (
+                    f'#% set name = "{config.name}" %}}\n'
+                    f'#% set version = "{config.version}" %}}\n'
+                )
```

Comment:
Instead of that, you can use

```python
set_global_jinja_var(recipe, "name", config.name)
set_global_jinja_var(recipe, "version", config.version)
```